specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/file.json

```diff
@@ -538,6 +526,62 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/fileServices/default/shares/{shareName}/lease": {
+      "post": {
+        "tags": [
+          "FileShares"
+        ],
+        "operationId": "FileShares_Lease",
+        "x-ms-examples": {
+          "Acquire a lease on a share": {
+            "$ref": "./examples/FileSharesLease_Acquire.json"
+          },
+          "Break a lease on a share": {
+            "$ref": "./examples/FileSharesLease_Break.json"
+          }
+        },
+        "description": "The Lease Share operation establishes and manages a lock on a share for delete operations. The lock duration can be 15 to 60 seconds, or can be infinite.",
+        "parameters": [
+          {
+            "$ref": "./storage.json#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "./storage.json#/parameters/StorageAccountName"
+          },
+          {
+            "$ref": "#/parameters/ShareName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "schema": {
+              "$ref": "#/definitions/LeaseShareRequest"
+            },
+            "description": "Lease Share request body."
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
Acquire lease return 204 (created), instead if 200.
Please make sure server and swagger behavior are aligned.

Reply:
I don't see 204 from API logs. I think it is 201 for acquire lease and 202 for break the lease. Will update the response code here.

Reply:
Yes, it should be 201. Please add it. and make sure all possible return codes are covered.